Working out vivarium-cell package

prismofeverything opened the pull request:
Making the adjustments to get the `vivarium-cell` package to use `vivarium-core`. All tests are passing now except the one in `death.py`, @U8NWXD maybe you have an idea about this one? It is asserting `saved_states` to equal `expected_saved_states`?

saved_states:

```
{'global': {
    'dead': [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1]},
 'cell': {
     'antibiotic': [0.0, 2.0, 4.0, 6.0, 8.0, 10.0, 12.0, 14.0, 16.0, 18.0, 20.0],
     'enduring_antibiotic': [0.0, 2.0, 4.0, 6.0, 8.0, 10.0, 12.0, 14.0, 16.0, 18.0, 20.0]},
 'time': [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]}
```

expected_saved_states:

```
{'global': {
     'dead': [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1]},
 'cell': {
    'antibiotic': [0.0, 2.0, 4.0, 6.0, 8.0, 8.0, 8.0, 8.0, 8.0, 8.0, 8.0],
    'enduring_antibiotic': [0.0, 2.0, 4.0, 6.0, 8.0, 10.0, 12.0, 14.0, 16.0, 18.0, 20.0]},
 'time': [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]}
```

Looks like the `cell`,`antibiotic` list tops out at 8.0 in the expected case while continuing to increase in the saved case. Thoughts?

Otherwise, we are good to go here. @eagmon I found a solution for adding things to the updater registry by calling `register` in `cell/__init__.py`, I think we can use a similar solution for derivers and processes. Let's talk about that at a future meeting.

prismofeverything committed:
working out cell package

U8NWXD commented:
@prismofeverything When the cell dies, the antibiotic injector process should get removed, causing the antibiotic variable to flat-line. That doesn't appear to be happening anymore. I'll look into why

U8NWXD commented:
@prismofeverything I think this problem occurs because `vivarium-core` is lacking the latest updates to `vivarium`. Specifically, I updated `update_in` to work with empty paths, which the death process relies on ([commit](https://github.com/CovertLab/vivarium/commit/ec49113815803bb8a4a46b6e21b996ab7c81b212#diff-ccfbb221f813a200ed620efba3cca839)). `vivarium-core` still has the old `update_in` ([link](https://github.com/vivarium-collective/vivarium-core/blob/master/vivarium/core/experiment.py#L83-L91)). I think updating `vivarium-core` with the latest `vivarium` changes will fix this

prismofeverything commented:
Hey @U8NWXD, thanks for taking a look! I agree, seems like there are some new changes to Vivarium since we made this migration. I'll pull those in.

prismofeverything commented:
Okay, that fixed the problem! All tests passing now.

U8NWXD commented on cell/processes/death.py:
```diff
@@ -351,4 +351,5 @@ def plot_death_freeze_state_test():
 
 
 if __name__ == '__main__':
+    test_death_freeze_state()
```
This looks like leftover debugging code?

prismofeverything replied:
It is indeed.

prismofeverything committed:
removed test

U8NWXD commented:
This looks good! I'll merge it once I confirm it works with the updated vivarium-core

U8NWXD committed:
Require vivarium-core >= 0.0.12

vivarium-cell's death process relies on an updated version of update_in
from core.experiment that was introduced in v0.0.12.

U8NWXD commented:
With version 0.0.12 of vivarium-core, the tests all pass. I updated setup.py to reflect that we need at least v0.0.12
